### PR TITLE
(RSFB-881)HASH function fix

### PR DESCRIPTION
**Pull request by Kowsikraja**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 1]
```diff
@@ -1558,6 +1558,40 @@ private SqlLibraryOperators() {
...
+          SqlFunctionCategory.SYSTEM);
+
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction SHA256 =
+      new SqlFunction("SHA256",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.explicit(SqlTypeName.VARCHAR)
+              .andThen(SqlTypeTransforms.TO_NULLABLE),
+          null,
+          OperandTypes.or(OperandTypes.family(SqlTypeFamily.STRING, SqlTypeFamily.INTEGER),
```
Comment (nosheenSheikh02) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1591]: You can use already existing STRING_INTEGER from OperandTypes.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -1558,6 +1558,40 @@ private SqlLibraryOperators() {
...
+              OperandTypes.family(SqlTypeFamily.BINARY, SqlTypeFamily.INTEGER)),
```
Comment (nosheenSheikh02) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1592]: You can create new Operand type in class OperandTypes => BINARY_INTEGER

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -10516,6 +10516,48 @@ public Sql schema(CalciteAssert.SchemaSpec schemaSpec) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
   }
 
+  @Test public void testSnowflakeHash() {
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:10519]: change this to testSnowflakeHashFunction

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -10516,6 +10516,48 @@ public Sql schema(CalciteAssert.SchemaSpec schemaSpec) {
...
+        .build();
+    final String expectedSFSql = "SELECT SHA2(\"ENAME\", 256) AS \"hashing\"\n"
+        + "FROM \"scott\".\"EMP\"";
+
+    assertThat(toSql(root, DatabaseProduct.SNOWFLAKE.getDialect()), isLinux(expectedSFSql));
+  }
+
+  @Test public void testSnowflakeSha256Function() {
+    final RelBuilder builder = relBuilder();
+    final RexNode sha2Node = builder.call(SqlLibraryOperators.SHA256,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:10549]: change this to sha256Node

**Review by nosheenSheikh02 — APPROVED** [review]

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**